erpnext/accounts/doctype/chart_of_accounts_importer/chart_of_accounts_importer.py

```diff
@@ -459,8 +459,14 @@ def unset_existing_data(company):
 		as_dict=True,
 	)
 
+	custom_linked = frappe.db.sql(
+		'''select fieldname from `tabCustom Field`
+		where fieldtype="Link" and options="Account" and dt="Company"''',
+		as_dict=True,
+	)
+
 	# remove accounts data from company
-	update_values = {d.fieldname: "" for d in linked}
+	update_values = {d.fieldname: "" for d in linked + custom_linked}
 	frappe.db.set_value("Company", company, update_values, update_values)
 
 	# remove accounts data from various doctypes
```

Comment:
This may be out of scope, but while we’re at it:
Do we want to use Database API here, too?
It would then trigger delete hooks and respect permissions, but normally, that should be the standard case.

```
for doctype in [
	"Account",
	"Party Account",
	"Mode of Payment Account",
	"Tax Withholding Account",
	"Sales Taxes and Charges Template",
	"Purchase Taxes and Charges Template",
]:
	frappe.delete_doc(doctype, filters={"company": company})
```

If permissions turn out to be a problem, you may want to do:
`frappe.delete_doc(doctype, filters={"company": company}, ignore_permissions=True)`
That‘s still safer than what we‘re doing now.